apps/studio/src/lib/db/clients/utils.ts

```diff
@@ -303,3 +305,43 @@ export function isAllowedReadOnlyQuery (identifiedQueries: IdentifyResult[], rea
 export const errorMessages = {
   readOnly: 'Write action(s) not allowed in Read-Only Mode.'
 }
+
+export async function getIAMPassword(awsProfile: string, region: string, hostname: string, port: number, username: string): Promise<string> {
+  const nodeProviderChainCredentials = fromIni({
+    profile: awsProfile ?? "default",
+  });
+  const signer = new Signer({
+    credentials: nodeProviderChainCredentials,
+    region,
+    hostname,
+    port,
+    username,
+  });
+  return  await signer.getAuthToken();
+}
+
+let resolvedPw: string | undefined;
+let tokenExpiryTime: number | null = null;
+
+export async function refreshTokenIfNeeded(redshiftOptions: any, server: any, port: number): Promise<string> {
+  if(!redshiftOptions?.iamAuthenticationEnabled){
+    return null
+  }
+
+  const now = Date.now();
+
+  if (!resolvedPw || !tokenExpiryTime || now >= tokenExpiryTime - 2 * 60 * 1000) { // Refresh 2 minutes before expiry
+    log.info("Refreshing IAM token...");
+    resolvedPw = await getIAMPassword(
+      redshiftOptions.awsProfile ?? "default",
+      redshiftOptions?.awsRegion,
+      server.config.host,
+      server.config.port || port,
+      server.config.user
+    );
+
+    tokenExpiryTime = now + 15 * 60 * 1000; // Tokens last 15 minutes
```

Comment:
please move these magic numbers go `globals`

Reply:
Added to globals

Reply:
it's still hard coded here